One crypto library

ronald-cron-arm opened the pull request:
## Description
Fixes https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/300
Last step in the move to a single crypto library.
It follows the merge of:
- https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/332
- https://github.com/Mbed-TLS/mbedtls-framework/pull/179
- First commit of https://github.com/Mbed-TLS/mbedtls/pull/10297

It cleans up what was done in https://github.com/Mbed-TLS/TF-PSA-Crypto/pull/332 to have a green CI before the changes done in mbedtls and mbedtls-framework.

## PR checklist
- [x] **changelog** not required because: the organization in core and driver libraries has not been released.
- [x] **framework PR** provided Mbed-TLS/mbedtls-framework#179
- [x] **mbedtls development PR** provided Mbed-TLS/mbedtls#10297  (first commit)
- [x] **mbedtls 3.6 PR** not required because: development specific 
- **tests**  not required because: clean-up

minosgalanakis reviewed (approved):
LGTM

bensze01 reviewed (approved):
LGTM

bensze01 reviewed (changes requested):
This PR [conflicts](https://mbedtls.trustedfirmware.org/blue/organizations/jenkins/mbed-tls-tf-psa-crypto-multibranch/detail/gh-readonly-queue%2Fdevelopment%2Fpr-385-19edaa785dd71ec8f0c9f72235243314c3d895fa/1/pipeline/1022) with `component_test_entropy_nv_seed_only`, added in #369. Please rebase on top of development, and update the paths in that component. https://github.com/Mbed-TLS/TF-PSA-Crypto/blob/19edaa785dd71ec8f0c9f72235243314c3d895fa/tests/scripts/components-configuration-platform.sh#L41-L64

ronald-cron-arm commented:
I've rebased on top of TF-PSA-Crypto/development as of 19edaa785dd71ec8f0c9f72235243314c3d895fa and updated component_test_entropy_nv_seed_only in an additional commit.

minosgalanakis reviewed (approved):
LGTM
